packages/fields/src/actions/permanently-delete-post.tsx

```diff
@@ -25,93 +34,153 @@ const permanentlyDeletePost: Action< PostWithPermissions > = {
 		const { status, permissions } = item;
 		return status === 'trash' && permissions?.delete;
 	},
-	async callback( posts, { registry, onActionPerformed } ) {
+	hideModalHeader: true,
+	RenderModal: ( { items, closeModal, onActionPerformed } ) => {
+		const [ isBusy, setIsBusy ] = useState( false );
 		const { createSuccessNotice, createErrorNotice } =
-			registry.dispatch( noticesStore );
-		const { deleteEntityRecord } = registry.dispatch( coreStore );
-		const promiseResult = await Promise.allSettled(
-			posts.map( ( post ) => {
-				return deleteEntityRecord(
-					'postType',
-					post.type,
-					post.id,
-					{ force: true },
-					{ throwOnError: true }
-				);
-			} )
+			useDispatch( noticesStore );
+		const { deleteEntityRecord } = useDispatch( coreStore );
+
+		return (
+			<VStack spacing="5">
+				<Text>
+					{ items.length > 1
+						? sprintf(
+								// translators: %d: number of items to delete.
+								_n(
+									'Permanently delete %d item?',
```

Comment:
Can we update the copy here to match other similar actions that start with: `Are you sure you want to...`?